### PR TITLE
update lifted files to k8s 1.26

### PR DESCRIPTION
**Pull request by zishen**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
For kamarda is based on k8s 1.26， but most lifted files are in 1.23.
So we need synchronize files to version 1.26 of k8s.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
update lifted files to k8s 1.26

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Review by RainbowMango — COMMENTED** [review]
/assign @XiShanYongYe-Chang

**Comment by zishen** [comment]
/cc @RainbowMango 
pls take a look.

**Comment by XiShanYongYe-Chang** [comment]
Thanks,  I will take a review ASAP.

File: pkg/util/lifted/deployment.go [hunk 1]
```diff
@@ -82,7 +79,7 @@ func ListReplicaSetsByDeployment(deployment *appsv1.Deployment, f ReplicaSetList
 	return owned, nil
 }
 
-// +lifted:source=https://github.com/kubernetes/kubernetes/blob/release-1.22/pkg/controller/deployment/util/deployment_util.go#L596-L628
+// +lifted:source=https://github.com/kubernetes/kubernetes/blob/release-1.26/pkg/controller/deployment/util/deployment_util.go#L563C1-L595
```
Comment (XiShanYongYe-Chang) [inline pkg/util/lifted/deployment.go:82]: A typo: L563C1

File: pkg/util/selector.go [hunk 2]
```diff
@@ -106,7 +106,7 @@ func ClusterMatches(cluster *clusterv1alpha1.Cluster, affinity policyv1alpha1.Cl
 
 	if affinity.FieldSelector != nil {
 		var matchFields labels.Selector
-		var err error
+		var err []error
```
Comment (XiShanYongYe-Chang) [inline pkg/util/selector.go:109]: err -> errs

File: pkg/util/lifted/objectwatcher.go [hunk 3]
```diff
@@ -84,5 +84,10 @@ func objectMetaObjEquivalent(a, b metav1.Object) bool {
 	if !reflect.DeepEqual(aLabels, bLabels) && (len(aLabels) != 0 || len(bLabels) != 0) {
 		return false
 	}
+	aAnnotations := a.GetAnnotations()
+	bAnnotations := b.GetAnnotations()
+	if !reflect.DeepEqual(aAnnotations, bAnnotations) && (len(aAnnotations) != 0 || len(bAnnotations) != 0) {
+		return false
+	}
```
Comment (XiShanYongYe-Chang) [inline pkg/util/lifted/objectwatcher.go:91]: This change may be affected by the logic. We can propose a PR separately and analyze the impact.

File: pkg/util/lifted/requestinfo.go [hunk 4]
```diff
@@ -161,11 +181,18 @@ func NewRequestInfo(req *http.Request) *apirequest.RequestInfo {
...
+			klog.ErrorS(err, "Couldn't parse request", "Request", req.URL.Query())
+			// Reset opts to not rely on partial results from parsing.
+			// However, if watch is set, let's report it.
+			opts = metainternalversion.ListOptions{}
+			if values := req.URL.Query()["watch"]; len(values) > 0 {
+				switch strings.ToLower(values[0]) {
+				case "false", "0":
+				default:
+					opts.Watch = true
+				}
```
Comment (XiShanYongYe-Chang) [inline pkg/util/lifted/requestinfo.go:195]: This change may be affected by the logic. We can propose a PR separately and analyze the impact.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks~

File: pkg/util/lifted/pubkeypin/pubkeypin.go [hunk 5]
```diff
@@ -56,18 +57,18 @@ func (s *Set) Allow(pubKeyHashes ...string) error {
 	for _, pubKeyHash := range pubKeyHashes {
 		parts := strings.Split(pubKeyHash, ":")
 		if len(parts) != 2 {
-			return fmt.Errorf("invalid hash, expected \"format:hex-value\". "+
+			return errors.Errorf("invalid hash, expected \"format:hex-value\". "+
```
Comment (RainbowMango) [inline pkg/util/lifted/pubkeypin/pubkeypin.go:60]: Please don't use `errors` as the package has been archived. `fmt` would be fine.

File: pkg/util/lifted/deployment.go (hunk 1 above)
Comment (zishen) [reply pkg/util/lifted/deployment.go:82]: > A typo: L563C1

yea, I will remove "C1"

File: pkg/util/selector.go (hunk 2 above)
Comment (zishen) [reply pkg/util/selector.go:109]: > err -> errs

OK

File: pkg/util/lifted/pubkeypin/pubkeypin.go (hunk 5 above)
Comment (zishen) [reply pkg/util/lifted/pubkeypin/pubkeypin.go:60]: > Please don't use `errors` as the package has been archived. `fmt` would be fine.

ok

File: pkg/util/lifted/objectwatcher.go (hunk 3 above)
Comment (zishen) [reply pkg/util/lifted/objectwatcher.go]: > This change may be affected by the logic. We can propose a PR separately and analyze the impact.

OK,  see another PR in #3792

File: pkg/util/lifted/requestinfo.go (hunk 4 above)
Comment (zishen) [reply pkg/util/lifted/requestinfo.go]: OK,  see another PR #3792

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Thanks
/lgtm

**Comment by zishen** [comment]
some thing go wrong.
image
Can you help me to retry it?
@RainbowMango

**Comment by RainbowMango** [comment]
done.

**Comment by zishen** [comment]
cc @RainbowMango It's ready to go.

**Review by RainbowMango — APPROVED** [review]
/approve

**Pull request merged**